editor/utils/url.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ */
+import { parse, format } from 'url';
+
+/**
+ * Appends arguments to the query string of the url
+ *
+ * @param  {String} url   URL
+ * @param  {Object} args  Query Args
+ *
+ * @return {String}       Updated URL
+ */
+export function addQueryArgs( url, args ) {
+	const parsedUrl = parse( url, true );
+	const query = { ...parsedUrl.query, ...args };
```

Comment:
Good point! that's my functional "everything is immutable" mind. (which is contradictory with the `delete` call right after)